src/esp/nav/PathFinder.cpp

```diff
@@ -1625,7 +1628,44 @@ PathFinder::Impl::getTopDownView(const float metersPerPixel,
   for (int h = 0; h < zResolution; ++h) {
     for (int w = 0; w < xResolution; ++w) {
       vec3f point = vec3f(curx, height, curz);
-      topdownMap(h, w) = isNavigable(point, 0.5);
+      topdownMap(h, w) = isNavigable(point, eps);
+      curx = curx + metersPerPixel;
+    }
+    curz = curz + metersPerPixel;
+    curx = startx;
+  }
+
+  return topdownMap;
+}
+
+typedef Eigen::Matrix<int, Eigen::Dynamic, Eigen::Dynamic> MatrixXi;
+
+MatrixXi PathFinder::Impl::getTopDownIslandView(const float metersPerPixel,
+                                                const float height,
+                                                const float eps) const {
+  std::pair<vec3f, vec3f> mapBounds = bounds();
+  vec3f bound1 = mapBounds.first;
+  vec3f bound2 = mapBounds.second;
+
+  float xspan = std::abs(bound1[0] - bound2[0]);
+  float zspan = std::abs(bound1[2] - bound2[2]);
+  int xResolution = xspan / metersPerPixel;
+  int zResolution = zspan / metersPerPixel;
```

Comment:
Do you want to truncate instead of rounding? 
```suggestion
  int xResolution = (xspan + 0.5) / metersPerPixel;
  int zResolution = (zspan + 0.5) / metersPerPixel;
```

Reply:
I don't think this is the behavior we'd want. This would add 0.5 meters to the navmesh bounds before dividing it into `metersPerPixel` sized cells. If we want to round, probably should use `math.ceil(xspan / metersPerPixel)` instead. In either case, this is the difference of 1 pixel. :shrug: